cairo/src/utils/dict.cairo

```diff
@@ -136,3 +137,83 @@ func hashdict_write{poseidon_ptr: PoseidonBuiltin*, dict_ptr: DictAccess*}(
     let dict_ptr = dict_ptr + DictAccess.SIZE;
     return ();
 }
+
+func hashdict_finalize{range_check_ptr}(
```

Comment:
maybe add some comments to explain the behaviors of `merge=FALSE` and `TRUE`

Reply:
eventually, I think that this function should be called `dict_update`, similarly to python `dict.update()` because it's what it does: updating the value of dict 1 (original_mapping) here with the values of dict 2 (dict_ptr here). For soundness, we need to add a `drop: bool`

Reply:
done